charts/karpenter/values.yaml

```diff
@@ -117,10 +117,20 @@ controller:
   # - name: aws-iam-token
   #   mountPath: /var/run/secrets/eks.amazonaws.com/serviceaccount
   #   readOnly: true
+
+  metrics:
+    # -- The container port to use for metrics.
+    port: 8080
+  healthProbe:
+    # -- The container port to use for http health probe.
+    port: 8081
```

Comment:
We try to minimize the API surface of the helm chart where possible. Instead, can this be wired through controller.env?

Reply:
adjusting controller.env is enough to avoid the conflict on launch and setup the listeners. However, since the ports are also declared in the [deployment](https://github.com/aws/karpenter/blob/7241f43569d6878056f3251667b4689684071401/charts/karpenter/templates/deployment.yaml#L88) for the purpose of liveness, [readiness](https://github.com/aws/karpenter/blob/7241f43569d6878056f3251667b4689684071401/charts/karpenter/templates/deployment.yaml#L107), [service](https://github.com/aws/karpenter/blob/7241f43569d6878056f3251667b4689684071401/charts/karpenter/templates/service.yaml#L17), and [servicemonitor ](https://github.com/aws/karpenter/blob/7241f43569d6878056f3251667b4689684071401/charts/karpenter/templates/servicemonitor.yaml#L24)access, the pod fails back into a CrashLoop as all checks fail as they are targeting the incorrect port.
I'm guessing this is also the reason behind the exposed value for `https-webhook` port.